WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/ScheduleToolbar.jsx

```diff
@@ -0,0 +1,190 @@
+// See https://github.com/react-bootstrap/react-bootstrap/issues/1345#issuecomment-142133819
+// for why we pass down ...props
+  return (
+    <Popover id="calendar-help-popover" title="Keyboard shortcuts help" {...props} >
+      <dl className="row">
+        <dt className="col-xs-4"><i className="fa fa-keyboard-o"/> or<br/> [C] + i</dt>
+        <dd className="col-xs-8">Toggle keyboard shortcuts</dd>
+        <dt className="col-xs-4">Arrow keys</dt>
+        <dd className="col-xs-8">Change selected event in calendar</dd>
+        <dt className="col-xs-4">[S] + Arrow keys</dt>
+        <dd className="col-xs-8">Change selected activity in picker</dd>
+        <dt className="col-xs-4">[Enter]</dt>
+        <dd className="col-xs-8">Add selected activity after selected event</dd>
+        <dt className="col-xs-4">[Del]</dt>
+        <dd className="col-xs-8">Remove selected event</dd>
+        <dt className="col-xs-4">[C] + Arrow keys</dt>
+        <dd className="col-xs-8">Move selected event around in calendar</dd>
+        <dt className="col-xs-4">[C] + [S] + up/down</dt>
+        <dd className="col-xs-8">Shrink/Expand selected event in calendar</dd>
+        <dt className="col-xs-4">[C] + [S] + click</dt>
+        <dd className="col-xs-8">Show contextual menu for event</dd>
+      </dl>
+      <hr />
+      <b>[C]:</b> ctrl key, <b>[S]:</b> shift key
+    </Popover>
+  );
+}
+
+const tooltipSettings = (
+  <Tooltip id="tooltip-calendar-settings">
+    Click to change the calendar's settings.
+  </Tooltip>
+);
+
+const TooltipKeyboard = ({ enabled, ...props }) => (
+  <Tooltip id="tooltip-enable-keyboard" {...props}>
+    Click to { enabled ? "disable" : "enable" } keyboard shortcuts
+  </Tooltip>
+);
+
+
+export const calendarOptionsInfo = {
+  slotDuration: {
+    label: "Minutes per row",
+    defaultValue: "00:30:00",
+    options: {
+      "5": "00:05:00",
+      "15": "00:15:00",
+      "20": "00:20:00",
+      "30": "00:30:00",
+    },
+  },
+  minTime: {
+    label: "Calendar starts at",
+    defaultValue: "8:00:00",
+    options: hours(),
+  },
+  maxTime: {
+    label: "Calendar ends at",
+    defaultValue: "20:00:00",
+    options: hours(),
+  },
+};
+
+const CalendarSettingsOption = ({selected, optionName, handlePropChange}) => {
+  let optionProps = calendarOptionsInfo[optionName];
+  return (
+    <Col xs={12}>
+      <Row>
+        <Col xs={6} className="setting-label">
+          {optionProps.label}
+        </Col>
+        <Col xs={6}>
+          <select className="form-control" value={selected} onChange={e => handlePropChange(optionName, e)}>
+            {_.map(optionProps.options, function(value, key) {
+              return (<option key={value} value={value}>{key}</option>)
+            })}
+          </select>
+        </Col>
+      </Row>
+    </Col>
+  );
+}
+
+function hours() {
+  let options = {};
+  for (let i = 0; i < 24; i++) {
+    options[i] = `${i}:00:00`;
+  }
+  return options;
+}
```

Comment:
Any reason for this to be a function instead of value? Also `for` loop likely indicates that lodash could come in handy, like so:
```js
const hours = _.fromPairs(_.times(24, i => [i, `${i}:00:00`]));
```